File: test_app_tz.js
const puppeteer = require('puppeteer');

(async () => {
    // Launch browser
    const browser = await puppeteer.launch({ headless: true, args: ['--no-sandbox'] });
    const page = await browser.newPage();

    // Listen to console to intercept our logged payload
    page.on('console', msg => console.log('BROWSER CONSOLE:', msg.text()));

    try {
        await page.goto('http://localhost:8548');
        await page.waitForSelector('input[name="username"]', { timeout: 5000 });
        
        // Use default admin credentials or previous ones
        await page.type('input[name="username"]', 'admin');
        await page.type('input[name="password"]', 'admin123'); // or whatever was set in DB reset
        await page.click('button[type="submit"]');

        await page.waitForSelector('table', { timeout: 5000 });
        
        // Click edit on the first trip
        await page.click('button[title="Edit"]');
        
        // Wait for modal
        await page.waitForSelector('input[type="datetime-local"]', { timeout: 5000 });
        
        // Find the log input explicitly and change it
        const inputs = await page.$$('input[type="datetime-local"]');
        if (inputs.length > 1) {
            // Usually the 2nd one is the first log's time
            await inputs[1].type(''); // clear it
            await inputs[1].type('0438PM'); // or just set value via evaluate to guarantee string
            await page.evaluate(() => {
                const els = document.querySelectorAll('input[type="datetime-local"]');
                if (els[1]) {
                    els[1].value = "2026-03-01T20:41";
                    // Dispatch event for React
                    els[1].dispatchEvent(new Event('input', { bubbles: true }));
                    els[1].dispatchEvent(new Event('change', { bubbles: true }));
                }
            });
        }

        // Save
        await page.evaluate(() => {
            const btns = Array.from(document.querySelectorAll('button'));
            const saveBtn = btns.find(b => b.textContent.includes('Save'));
            if (saveBtn) saveBtn.click();
        });

        // Wait a sec for the payload to log
        await new Promise(r => setTimeout(r, 2000));
        
    } catch (e) {
        console.error('Test Error:', e);
    } finally {
        await browser.close();
    }
})();

File: test_app_tz.py
import requests

base_url = "http://localhost:8003"
resp = requests.post(f"{base_url}/auth/token", data={"username": "admin", "password": "admin123"})
token = resp.json()["access_token"]
headers = {"Authorization": f"Bearer {token}"}

trips = requests.get(f"{base_url}/admin/trips", headers=headers).json()
for t in trips:
    if t["logs"]:
        trip = t
        break
else:
    print("NO TRIPS WITH LOGS IN DB")
    exit(1)

trip_id = trip["id"]
logs = trip["logs"]
print("BEFORE Logs:", [(l["id"], l["state"], l["timestamp"]) for l in logs])

log_updates = []
for l in logs:
    log_updates.append({
        "id": l["id"], 
        "timestamp": "2026-03-01T20:41", # String precisely from datetime-local input
        "address": l["address"], 
        "state": l["state"]
    })

payload = {
    "driver_id": trip["driver_id"],
    "status": trip["status"],
    "start_date": trip["start_date"],
    "logs": log_updates
}

ret = requests.put(f"{base_url}/admin/trips/{trip_id}", json=payload, headers=headers)
if ret.status_code != 200:
    print("Error:", ret.status_code, ret.text)
else:
    updated = ret.json()
    print("AFTER Logs:", [(l["id"], l["state"], l["timestamp"]) for l in updated.get("logs", [])])
